core/src/main/java/nu/yona/server/goals/service/GoalDTO.java

```diff
@@ -56,6 +54,11 @@ public boolean isMandatory()
 
 	public abstract Goal createGoalEntity();
 
+	protected static UUID determineActivityCategoryID(String activityCategoryUrl)
+	{
+		return UUID.fromString(activityCategoryUrl.substring(activityCategoryUrl.lastIndexOf('/') + 1));
```

Comment:
Sure, but then the DTO needs two properties, one for the URL and one for the ID. This way, it's parsed at the time it's received, so you don't need to keep it.